### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/snansumors`

### PR DESCRIPTION
**Pull request by Snehil-Shah**
## Description

> What is the purpose of this pull request?

This pull request:

- adds C ndarray API for blas/ext/base/snansumpw.
- refactors package to follow updated project conventions.
- Updates stride handling where stride is equal to zero.
- Adds test cases for NaN values.

## Related Issues

> Does this pull request have any related issues?

No.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/snansumors/README.md [hunk 1]
```diff
@@ -171,7 +285,7 @@ console.log( v );
...
 -   <span class="package-name">[`@stdlib/blas/ext/base/snansumpw`][@stdlib/blas/ext/base/snansumpw]</span><span class="delimiter">: </span><span class="description">calculate the sum of single-precision floating-point strided array elements, ignoring NaN values and using pairwise summation.</span>
--   <span class="package-name">[`@stdlib/blas/ext/base/ssumors`][@stdlib/blas/ext/base/ssumors]</span><span class="delimiter">: </span><span class="description">calculate the sum of single-precision floating-point strided array elements using ordinary recursive summation.</span>
+-   <span class="package-name">[`@stdlib/blas/ext/base/snansumors`][@stdlib/blas/ext/base/snansumors]</span><span class="delimiter">: </span><span class="description">calculate the sum of single-precision floating-point strided array elements using ordinary recursive summation.</span>
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/snansumors/README.md]:
```suggestion
-   <span class="package-name">[`@stdlib/blas/ext/base/ssumors`][@stdlib/blas/ext/base/snansumors]</span><span class="delimiter">: </span><span class="description">calculate the sum of single-precision floating-point strided array elements using ordinary recursive summation.</span>
```

File: lib/node_modules/@stdlib/blas/ext/base/snansumors/README.md [hunk 2]
```diff
@@ -197,7 +311,7 @@ console.log( v );
 
 [@stdlib/blas/ext/base/snansumpw]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/snansumpw
 
-[@stdlib/blas/ext/base/ssumors]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/ssumors
+[@stdlib/blas/ext/base/snansumors]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/snansumors
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/snansumors/README.md]:
```suggestion
[@stdlib/blas/ext/base/ssumors]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/ssumors
```

File: lib/node_modules/@stdlib/blas/ext/base/snansumors/docs/repl.txt [hunk 3]
```diff
@@ -34,7 +34,7 @@
     > {{alias}}( x.length, x, 1 )
     1.0
 
-    // Using `N` and `stride` parameters:
+    // Using `N` and stride parameters:
     > x = new {{alias:@stdlib/array/float32}}( [ -2.0, 1.0, 1.0, -5.0, 2.0, -1.0, NaN, NaN ] );
     > var N = 4;
     > var stride = 2;
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/snansumors/docs/repl.txt]: The use of `var N` can be removed from here and below.

File: lib/node_modules/@stdlib/blas/ext/base/snansumors/README.md [hunk 4]
```diff
@@ -36,29 +36,27 @@ limitations under the License.
...
 -   **N**: number of indexed elements.
 -   **x**: input [`Float32Array`][@stdlib/array/float32].
--   **stride**: index increment for the strided array.
+-   **strideX**: stride length for `x`.
 
-The `N` and `stride` parameters determine which elements in the strided array are 
-accessed at runtime. For example, to compute the sum of every other element in 
-the strided array.
+The `N` and stride parameters determine which elements in the strided array are 
+accessed at runtime. For example, to compute the sum of every other element:
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/snansumors/README.md]:
```suggestion
The `N` and stride parameters determine which elements in the strided array are  accessed at runtime. For example, to compute the sum of every other element:
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Snehil-Shah and @headlessNode!

**Pull request merged**